components/Form.jsx

```diff
@@ -17,7 +18,29 @@ export default function Form({ onSubmit, initialValues = {} }) {
 
     // Custom validation: at least one fertiliserSeason checkbox checked
     if (data.fertiliserSeason.length === 0) {
-      alert("Please select at least one fertiliser season.");
+      toast.error("Please select at least one fertiliser season.", {
+        position: "top-center",
+      });
+      return;
+    }
+
+    try {
+      new URL(data.imageUrl);
+    } catch {
+      toast.error("Please enter a valid URL (including https//...)", {
+        position: "top-center",
+      });
+
+      return;
+    }
+
+    const imagePattern = /\.(jpeg|jpg|gif|png|webp)$/i;
```

Comment:
```suggestion
    const imagePattern = /\.(jpeg|jpg|gif|png|webp)(\?.*)?$/i;
```

Cloudinary URLs almost always will come with some query parameters or long path strings that include some kind of transformations like:

`https://res.cloudinary.com/yourcloud/image/upload/myimage.jpg?auth=xyz`

So this ("only match if the string ends exactly with .jpg, .jpeg, etc") works for now since users are just pasting image URLs but once you implement image upload you’ll be generating them.

Reply:
Just doublechecked: Cloudinary URLs typically don't have query parameters unless you're using signed URLs or secure delivery features. So your simple regex is fine for both now and your upcoming upload based setup! ✅